pyogrio/tests/test_path.py

```diff
@@ -49,6 +50,20 @@ def change_cwd(path):
             "zip+https://s3.amazonaws.com/testing/shapefile.zip",
             "/vsizip/vsicurl/https://s3.amazonaws.com/testing/shapefile.zip",
         ),
+        # auto-prefix zip files
+        ("test.zip", "/vsizip/test.zip"),
+        ("/a/b/test.zip", "/vsizip//a/b/test.zip"),
+        ("a/b/test.zip", "/vsizip/a/b/test.zip"),
+        # archives using ! notation should be prefixed by vsizip
+        ("test.zip!item.shp", "/vsizip/test.zip/item.shp"),
+        ("test.zip!/a/b/item.shp", "/vsizip/test.zip/a/b/item.shp"),
+        ("test.zip!a/b/item.shp", "/vsizip/test.zip/a/b/item.shp"),
+        ("/vsizip/test.zip/a/b/item.shp", "/vsizip/test.zip/a/b/item.shp"),
+        ("zip:///test.zip/a/b/item.shp", "/vsizip//test.zip/a/b/item.shp"),
+        (
+            "zip+https://s3.amazonaws.com/fiona-testing/coutwildrnp.zip",
```

Comment:
```suggestion
            "https://s3.amazonaws.com/fiona-testing/coutwildrnp.zip",
```

Does this also work? (and the version of zip+https is already included just above the new code)

Reply:
It uncovered an error!  Great!  We needed to put the `zip` scheme first rather than just appending it, otherwise the `https://` scheme got clobbered.  Fixed and added more tests.